src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -27,8 +29,117 @@ interface Props {
   filtersVisible: boolean;
   setFiltersVisible: (boolean) => void;
   user: User,
+  downloadInProgress: boolean,
+  setDownloadInProgress: (boolean) => void,
+  downloadStatusMessage: {
+    title: string,
+    message: string,
+    active: boolean
+  },
+  setDownloadStatusMessage: (Object) => void;
 }
 
+const handleDownloadZipClick = async (
+  selectedResources: any[],
+  setDownloadInProgress: (boolean) => void,
+  setDownloadStatusMessage: (object) => void,
+  manifestFieldName: string,
+  maxDownloadSizeBytes: number,
+) => {
+  const studyIDs = selectedResources.map((study) => study.project_number);
+  let downloadSize = 0;
+  selectedResources.forEach(
+    (study) => {
+      study[manifestFieldName].forEach(
+        (file) => {
+          downloadSize += file.file_size;
```

Comment:
What if that entry don't have `file_size`? should we query indexd?

In fact, I was hope all this file size calculation and limitation thing would happen in the backend job, and Portal as a frontend component just get whatever the messgaes from the backend job, like "error the file size is too large"

I think that is consistent with your design doc's API design section

Reply:
There is also a check for this in the backend job: https://github.com/uc-cdis/sower-jobs/blob/1a922a7698e2e44e0f537141e3adf0de9db6b7f1/export_job/export_zip_from_manifest_job.py#L64

I added this additional check here because it would take a while for the user to get the error back from the job response, by comparison to just adding this extra check. 

But yes, I'll add checks to make sure `__manifest` and `file_size` are present too

Reply:
If BE and FE are doing the same thing, I would perfer to relief that duty from our FE compoment

In this scenario, after submitting the job, the compont should just trun into a `pending` state, and wait for the response from BE to figure out what to present to the user

Reply:
And: if a study don't have this `file_size` info in the manifest, should it be remove from the ID list that sends to the job?

If not, will there be discrepancy? Like the FE thinks the user should be able to download the file according to the manifest data, and BE realizes the total size is actually out of limit?